cudax/include/cuda/experimental/__memory_resource/cuda_async_memory_resource.h

```diff
@@ -0,0 +1,294 @@
+      &__ptr,
+      __bytes,
+      __pool_,
+      __stream.get());
+    return __ptr;
+  }
+
+  //! @brief Deallocate memory pointed to by \p __ptr.
+  //! @param __ptr Pointer to be deallocated. Must have been allocated through a call to `allocate_async`
+  //! @param __bytes The number of bytes that was passed to the `allocate_async` call that returned \p __ptr.
+  //! @param __alignment The alignment that was passed to the `allocate_async` call that returned \p __ptr.
+  //! @param __stream A stream that has a stream ordering relationship with the stream used in the
+  //! <a href="https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY__POOLS.html">allocate_async</a> call
+  //! that returned \p __ptr.
+  void deallocate_async(void* __ptr, const size_t __bytes, const size_t __alignment, const ::cuda::stream_ref __stream)
+  {
+    // We need to ensure that the provided alignment matches the minimal provided alignment
+    _LIBCUDACXX_ASSERT(__is_valid_alignment(__alignment),
+                       "Invalid alignment passed to cuda_async_memory_resource::deallocate.");
+    deallocate_async(__ptr, __bytes, __stream);
+    (void) __alignment;
+  }
+
+  //! @brief Deallocate memory pointed to by \p __ptr.
+  //! @param __ptr Pointer to be deallocated. Must have been allocated through a call to `allocate_async`
+  //! @param __bytes The number of bytes that was passed to the `allocate_async` call that returned \p __ptr.
+  //! @param __stream A stream that has a stream ordering relationship with the stream used in the <a
+  //! href="https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY__POOLS.html">allocate_async</a> call
+  //! that returned \p __ptr.
+  void deallocate_async(void* __ptr, size_t, const ::cuda::stream_ref __stream)
+  {
+    _CCCL_ASSERT_CUDA_API(::cudaFreeAsync, "cuda_async_memory_resource::deallocate_async failed", __ptr, __stream.get());
+  }
+
+  //! @brief Equality comparison with another cuda_async_memory_resource
+  //! @return true if underlying \c cudaMemPool_t are equal
+  _CCCL_NODISCARD constexpr bool operator==(cuda_async_memory_resource const& __rhs) const noexcept
+  {
+    return __pool_ == __rhs.__pool_;
+  }
+#    if _CCCL_STD_VER <= 2017
+
+  //! @brief Inequality comparison with another cuda_async_memory_resource
+  //! @return true if underlying \c cudaMemPool_t are inequal
+  _CCCL_NODISCARD constexpr bool operator!=(cuda_async_memory_resource const& __rhs) const noexcept
+  {
+    return __pool_ != __rhs.__pool_;
+  }
+#    endif // _CCCL_STD_VER <= 2017
+
+  //! @brief Equality comparison between a cuda_async_memory_resource and another resource
+  //! @param __lhs The cuda_async_memory_resource
+  //! @param __rhs The resource to compare to
+  //! @return If the underlying types are equality comparable, returns the result of equality comparison of both
+  //! resources. Otherwise, returns false.
```

Comment:
```suggestion
  //! @brief Equality comparison between a cuda_async_memory_resource and another resource.
  //! @param __lhs The cuda_async_memory_resource.
  //! @param __rhs The resource to compare to.
  //! @returns If the underlying types are equality comparable, returns the result of equality comparison of both
  //! resources. Otherwise, returns false.
```